src/compose.ts

```diff
@@ -158,30 +162,39 @@ function normalizeObjectToComposition(
 			c = { version: SchemaVersion.v2_0, services: c };
 		// FIXME: perform attribute migration
 		case SchemaVersion.v2_0:
-			c.version = SchemaVersion.v2_1;
+			c.version = SchemaVersion.v2_4;
 		/* no attributes migration needed for 2.0->2.1 */
 		case SchemaVersion.v2_1:
+			c.version = SchemaVersion.v2_4;
+		/* no attributes migration needed for 2.1->2.4 */
+		case SchemaVersion.v2_4:
```

Comment:
Again, what happens with `2.2` and `2.3`? Aren't these acceptable and don't we need to transform input? Ie. are those versions a strict subset of `2.4`?

Reply:
:+1: they should be subsets of `2.4`